src/databricks/labs/ucx/source_code/base.py

```diff
@@ -113,3 +113,8 @@ def __init__(self, linters: list[Linter]):
     def lint(self, code: str) -> Iterable[Advice]:
         for linter in self._linters:
             yield from linter.lint(code)
+
+
+@dataclass
+class LocatedAdvice(Advice):
```

Comment:
and adding `def for_path(self, path: Path) -> LocatedAdvice:` on `Advice` would be simplifying as well.

Reply:
done